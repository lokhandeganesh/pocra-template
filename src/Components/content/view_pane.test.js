import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import ViewPane from './view_pane';


test('checks correct response to tab-selection in the view-pane', () => {
	const { container, getByText } = render(<ViewPane />);
	const tabs_panes_grid = container.querySelector('#content-tabs-panes-grid');
	const maps_tab_button_grid = container.querySelector('#maps-tab-button-grid');
	const data_tab_button_grid = container.querySelector('#data-tab-button-grid');

	fireEvent.click(getByText('Maps'));
	expect(maps_tab_button_grid.className === 'selected');
	expect(data_tab_button_grid.className === 'not-selected');
	// expect(tabs_panes_grid).toMatchSnapshot();

	fireEvent.click(getByText('Data'));
	expect(maps_tab_button_grid.className === 'not-selected');
	expect(data_tab_button_grid.className === 'selected');
	// expect(tabs_panes_grid).toMatchSnapshot();
});