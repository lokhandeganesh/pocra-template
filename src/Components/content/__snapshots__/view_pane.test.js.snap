// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks correct response to tab-selection in the view-pane 1`] = `
<div
  id="content-tabs-panes-grid"
>
  <div>
    Maps Div
  </div>
</div>
`;

exports[`checks correct response to tab-selection in the view-pane 2`] = `
<div
  id="content-tabs-panes-grid"
>
  <div>
    Data Div
  </div>
</div>
`;
